Cubical/ZCohomology/CohomologyRings/S2wedgeS4.agda

```diff
@@ -0,0 +1,522 @@
+open import Cubical.ZCohomology.CohomologyRings.CupProductProperties
+
+open Iso
+
+
+module Equiv-RP2-Properties where
+
+-----------------------------------------------------------------------------
+-- Definitions, Import with notations, Partition
+
+  -- Definition
+  private
+    ℤAG = Ring→AbGroup (CommRing→Ring ℤCR)
+
+  <XY,X²,Y²> : FinVec ℤ[x,y] 3
+  <XY,X²,Y²> zero = base (1 ∷ 1 ∷ []) 1
+  <XY,X²,Y²> one = base (2 ∷ 0 ∷ []) 1
+  <XY,X²,Y²> two = base (0 ∷ 2 ∷ []) 1
+
+  ℤ[X,Y]/<XY,X²,Y²> : CommRing ℓ-zero
+  ℤ[X,Y]/<XY,X²,Y²> = PolyCommRing-Quotient ℤCR <XY,X²,Y²>
+
+  ℤ[x,y]/<xy,x²,y²> : Type ℓ-zero
+  ℤ[x,y]/<xy,x²,y²> = fst ℤ[X,Y]/<XY,X²,Y²>
+
+  -- Import with notation
+  open IsGroupHom
+  open gradedRingProperties
+
+  open CommRingStr (snd ℤCR) using ()
+    renaming
+    ( 0r        to 0ℤ
+    ; 1r        to 1ℤ
+    ; _+_       to _+ℤ_
+    ; -_        to -ℤ_
+    ; _·_       to _·ℤ_
+    ; +Assoc    to +ℤAssoc
+    ; +IdL      to +ℤIdL
+    ; +IdR      to +ℤIdR
+    ; +Comm     to +ℤComm
+    ; ·Assoc    to ·ℤAssoc
+    ; ·IdL      to ·ℤIdL
+    ; ·IdR      to ·ℤIdR
+    ; ·DistR+   to ·ℤDistR+
+    ; ·Comm     to ·ℤComm
+    ; is-set    to isSetℤ     )
+
+  open RingStr (snd (H*R S²⋁S⁴)) using ()
+    renaming
+    ( 0r        to 0H*
+    ; 1r        to 1H*
+    ; _+_       to _+H*_
+    ; -_        to -H*_
+    ; _·_       to _cup_
+    ; +Assoc    to +H*Assoc
+    ; +IdL      to +H*IdL
+    ; +IdR      to +H*IdR
+    ; +Comm     to +H*Comm
+    ; ·Assoc    to ·H*Assoc
+    ; ·IdL      to ·H*IdL
+    ; ·IdR      to ·H*IdR
+    ; ·DistR+   to ·H*DistR+
+    ; is-set    to isSetH*     )
+
+  open CommRingStr (snd ℤ[X,Y]) using ()
+    renaming
+    ( 0r        to 0Pℤ
+    ; 1r        to 1Pℤ
+    ; _+_       to _+Pℤ_
+    ; -_        to -Pℤ_
+    ; _·_       to _·Pℤ_
+    ; +Assoc    to +PℤAssoc
+    ; +IdL      to +PℤIdL
+    ; +IdR      to +PℤIdR
+    ; +Comm     to +PℤComm
+    ; ·Assoc    to ·PℤAssoc
+    ; ·IdL      to ·PℤIdL
+    ; ·IdR      to ·PℤIdR
+    ; ·Comm     to ·PℤComm
+    ; ·DistR+   to ·PℤDistR+
+    ; is-set    to isSetPℤ     )
+
+  open CommRingStr (snd ℤ[X,Y]/<XY,X²,Y²>) using ()
+    renaming
+    ( 0r        to 0PℤI
+    ; 1r        to 1PℤI
+    ; _+_       to _+PℤI_
+    ; -_        to -PℤI_
+    ; _·_       to _·PℤI_
+    ; +Assoc    to +PℤIAssoc
+    ; +IdL      to +PℤIIdL
+    ; +IdR      to +PℤIIdR
+    ; +Comm     to +PℤIComm
+    ; ·Assoc    to ·PℤIAssoc
+    ; ·IdL      to ·PℤIIdL
+    ; ·IdR      to ·PℤIIdR
+    ; ·DistR+   to ·PℤIDistR+
+    ; is-set    to isSetPℤI     )
+
+
+  e₀ = invGroupIso H⁰-S²⋁S⁴≅ℤ
+  ϕ₀ = fun (fst e₀)
+  ϕ₀str = snd e₀
+  ϕ₀⁻¹ = inv (fst e₀)
+  ϕ₀⁻¹str = snd (invGroupIso e₀)
+  ϕ₀-sect = rightInv (fst e₀)
+  ϕ₀-retr = leftInv (fst e₀)
+
+  e₂ = invGroupIso H²-S²⋁S⁴≅ℤ
+  ϕ₂ = fun (fst e₂)
+  ϕ₂str = snd e₂
+  ϕ₂⁻¹ = inv (fst e₂)
+  ϕ₂⁻¹str = snd (invGroupIso e₂)
+  ϕ₂-sect = rightInv (fst e₂)
+  ϕ₂-retr = leftInv (fst e₂)
+
+  module WhyWontYouWork
```

Comment:
This module name isn't great

Reply:
Alternatively document it

Reply:
Well I was hoping to find a solution, but couldn't.
I changed the name and added a comment